client/src/services/authorizeUser.js

```diff
@@ -0,0 +1,48 @@
+import getAPI from "./util";
+import App from "App";
+
+// var promiseObj = new Promise(executor);
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise
+let promiseObjAuthUser = new Promise(authorizeUser(authToken))
```

Comment:
This is cool! `fetch` automatically returns a promise, so I think you can just return `fetch`. It'll look something like this:

```js
// User.js
const authorizeUser = (authToken) => {
  return fetch(`${API_URL}/auth/user`, {
    headers: {
      Authorization: `Token ${localStorage.getItem("token")}`,
    },
  })
```

Then, in the login page:

```js
// pages/Login.js
import authroizeUser from ../services/User.js 

// ...

componentDidMount() {
	authorizeUser.then(json => {
		// whatever, probably set state
	})
}
```

This wont be exactly what you need, but it should be close